ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/agent/MLAgentExecutor.java

```diff
@@ -124,28 +132,27 @@ public void execute(Input input, ActionListener<Output> listener) {
                                     (ConversationIndexMemory.Factory) memoryFactoryMap.get(memorySpec.getType());
                                 conversationIndexMemoryFactory.create(question, memoryId, appType, ActionListener.wrap(memory -> {
                                     inputDataSet.getParameters().put(MEMORY_ID, memory.getConversationId());
-                                    // Create root interaction ID
-                                    ConversationIndexMessage msg = ConversationIndexMessage
-                                        .conversationIndexMessageBuilder()
-                                        .type(appType)
-                                        .question(question)
-                                        .response("")
-                                        .finalAnswer(true)
-                                        .sessionId(memory.getConversationId())
-                                        .build();
-                                    memory.save(msg, null, null, null, ActionListener.<CreateInteractionResponse>wrap(interaction -> {
-                                        log.info("Created parent interaction ID: " + interaction.getId());
-                                        inputDataSet.getParameters().put(PARENT_INTERACTION_ID, interaction.getId());
-                                        ActionListener<Object> agentActionListener = createAgentActionListener(
-                                            listener,
-                                            outputs,
-                                            modelTensors
-                                        );
-                                        executeAgent(inputDataSet, mlAgent, agentActionListener);
-                                    }, ex -> {
-                                        log.error("Failed to create parent interaction", ex);
-                                        listener.onFailure(ex);
-                                    }));
+                                    ActionListener<Object> agentActionListener = createAgentActionListener(listener, outputs, modelTensors);
+                                    // get question for regenerate
+                                    if (regenerateInteractionId != null) {
+                                        log.info("Regenerate for existing interaction {}", regenerateInteractionId);
+                                        client
```

Comment:
Should we have the question absence check before this client execution? So if inputDataSet has question parameter, we don't need to call client, right?

Reply:
call client do two things here
 1. get original question 
 2. validate the interaction is exist